neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/dialect/impl/Neo4jEntityQueries.java

```diff
@@ -54,6 +57,62 @@ public Neo4jEntityQueries(EntityKeyMetadata entityKeyMetadata) {
 		this.removeEntityQuery = initRemoveEntityQuery( entityKeyMetadata );
 		this.updateEmbeddedNodeQuery = initUpdateEmbeddedNodeQuery( entityKeyMetadata );
 		this.removeEmbdeddedElementQuery = initRemoveEmbdeddedElementQuery( entityKeyMetadata );
+		this.findAssociationPartialQuery = initFindAssociationPartialQuery( entityKeyMetadata );
+	}
+
+	/*
+	 * Example:
+	 *
+	 * MATCH (owner:ENTITY:AccountOwner {id: {0}})
+	 */
+	private static String initFindAssociationPartialQuery(EntityKeyMetadata entityKeyMetadata) {
+		StringBuilder queryBuilder = new StringBuilder( "MATCH " );
+		queryBuilder.append( "(owner:" );
+		queryBuilder.append( ENTITY );
+		queryBuilder.append( ":" );
+		appendLabel( entityKeyMetadata, queryBuilder );
+		appendProperties( entityKeyMetadata, queryBuilder );
+		queryBuilder.append( ")" );
+		return queryBuilder.toString();
+	}
+
+	/*
+	 * Example:
+	 *
+	 * MATCH (owner:ENTITY:Car {`carId.maker`: {0}, `carId.model`: {1}}) -[r:tires]- ()
+	 * RETURN r
+	 *
+	 * or for embedded associations:
+	 *
+	 * MATCH (owner:ENTITY:StoryGame {id: {0}}) -[:evilBranch]-> (:EMBEDDED) -[r:additionalEndings]-> (:EMBEDDED)
+	 * RETURN r
+	 */
+	private String completeFindAssociationQuery(String relationshipType) {
```

Comment:
Initially I've created it there but then I've moved it here because sometimes you need the inverse association of the one you are considering and this query depends mainly on the entityKeyMetadata.

Now that I think about it I could see if I can create a function like findAssociationOnTheMainSide(..) when I have the associationKey.

I'll have a look